.expeditor/Version_inspector/SW-Version-update.py

```diff
@@ -0,0 +1,202 @@
+from math import prod
+import requests
+from bs4 import BeautifulSoup
+import re
+from packaging import version
+import json
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+
+pf = open('prod_list.json')
+mf = open('manual_prod_list.json')
+pvf = open('prod_default_version.json')
+
+
+file_data_pf = json.load(pf)
+file_data_mf = json.load(mf)
+file_data_pvf = json.load(pvf)
+
+prod_lst=list(file_data_pf.keys()) #  to validate the product 
+manual_prod_list=list(file_data_mf.keys())
+prod_def_ver_list=list(file_data_pvf.keys())
+
+choice = int(input("Enter \n1 - for Latest version check for all products.\n2 - Check for a particular S/W \n3 - List all S/W that needs to be handled manually \n"))
+
+def get_latest_version(url,expr,product):
+    response = requests.get(url)
+    soup = BeautifulSoup(response.content, 'html.parser')
+    
+
+    for data in soup(['style', 'script']):
+        data.decompose()
+
+    list_of_contents = ' '.join(soup.stripped_strings).split(" ")
+    
+    r = re.compile(expr)
+    filtered_list = list(filter(r.match, list_of_contents)) # Read Note
+    
+    print("======================================================")
+    versions_string = " ".join(filtered_list)
+
+    if product in ["server-open-jre"]:
+        extracted_versions = re.findall(r"\d[\.\d]*\d_\d+", versions_string)
+    elif(product in ["libedit"]):
+        extracted_versions = re.findall(r"\d+-[\.\d]*\d", versions_string)
+    else:
+        extracted_versions = re.findall(r"\d[\.\d]*\d", versions_string)
+    #print(extracted_versions)
+    
+    highest_version = extracted_versions[0]
+    for i in range(1,len(extracted_versions)):
+    #print("Comparing ",extracted_versions[i]," and ",highest_version)
+        if(version.parse(extracted_versions[i]) > version.parse(highest_version)):
+            highest_version = extracted_versions[i]
+    return highest_version
+
+def mail_sender(mail_content):
+    sender_address = ''
+    sender_pass = ''
+    receiver_address = ''
+    #Setup the MIME
+    message = MIMEMultipart()
+    message['From'] = sender_address
+    message['To'] = receiver_address
+    message['Subject'] = 'Version Updates Automation!'   #The subject line
+    #The body and the attachments for the mail
+    message.attach(MIMEText(mail_content, 'plain'))
+    #Create SMTP session for sending the mail
+    session = smtplib.SMTP('smtp.gmail.com',465) #use gmail with port
+    session.starttls() #enable security
+    session.login(sender_address, sender_pass) #login with mail_id and password
+    text = message.as_string()
+    session.sendmail(sender_address, receiver_address, text)
+    session.quit()
+    print('Mail Sent')
+
+
```

Comment:
`# TODO - Update this method to work with script arguments` or something similar can be mentioned for future reference